streams/jvm/src/main/scala/scalaz/zio/stream/platform.scala

```diff
@@ -0,0 +1,57 @@
+package scalaz.zio.stream
+
+import java.io.{ IOException, InputStream, OutputStream }
+
+import scalaz.zio._
+import scalaz.zio.blocking._
+
+trait ZStreamPlatformSpecific {
+
+  /**
+   * Uses the provided `TaskR` value to create a [[ZStream]] of byte chunks, backed by
+   * the resulting `InputStream`. When data from the `InputStream` is exhausted,
+   * the stream will close it.
+   */
+  def fromInputStream[R](
+    is: TaskR[R, InputStream],
+    chunkSize: Int = ZStreamChunk.DefaultChunkSize
+  ): ZStreamChunk[R with Blocking, Throwable, Byte] =
```

Comment:
This one still returns `Throwable`, can you change to `IOException` with `refineOrDie` below?

Reply:
Hmm but that means that anyone passing the InputStream has to pass it with a task that can only fail with that exception.

Reply:
Maybe we should just take a pure InputStream? That passes the onus of managing it to the user but maybe it's fine for starters.

Reply:
Ah, you're right, sorry, missed that.

Reply:
Hmm, actually, why not just accept `InputStream` instead of `TaskR` of InputStream? Usually accepting a task like this when not building a combinator is not needed.

Reply:
Yeah let's do that.